server/server.go

```diff
@@ -179,24 +179,29 @@ func (s *server) buildInvokeChain() {
 
 // RegisterService should not be called by users directly.
 func (s *server) RegisterService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) error {
-	if s.svcInfo != nil {
-		panic(fmt.Sprintf("Service[%s] is already defined", s.svcInfo.ServiceName))
+	if svcInfo == nil {
+		return errors.New("svcInfo is nil. please specify non-nil svcInfo")
```

Comment:
I think no need to judge whether svcInfo or handler is nil, just panic if user pass a nil svcInfo or handler.

Reply:
Hmm I agree with you